examples/point_discharge/point_discharge.py

```diff
@@ -0,0 +1,261 @@
+    2D hydrodynamics TELEMAC-2D software release 7.0 user
+    manual." Paris:  R&D, Electricite de France, p. 134
+    (2014).
+
+[2] J.G. Wallwork, N. Barral, D.A. Ham, M.D. Piggott,
+    "Goal-Oriented Error Estimation and Mesh Adaptation for
+    Tracer Transport Modelling", submitted to Computer
+    Aided Design (2021).
+
+[3] B.P. Flannery, W.H. Press, S.A. Teukolsky, W. Vetterling,
+    "Numerical recipes in C", Press Syndicate of the University
+    of Cambridge, New York (1992).
+"""
+from thetis import *
+import numpy as np
+
+
+def bessi0(x):
+    """
+    Modified Bessel function of the first kind. Code taken from [3].
+    """
+    ax = abs(x)
+    y1 = x/3.75
+    y1 *= y1
+    expr1 = 1.0 + y1*(3.5156229 + y1*(3.0899424 + y1*(1.2067492 + y1*(
+        0.2659732 + y1*(0.360768e-1 + y1*0.45813e-2)))))
+    y2 = 3.75/ax
+    expr2 = exp(ax)/sqrt(ax)*(0.39894228 + y2*(0.1328592e-1 + y2*(
+        0.225319e-2 + y2*(-0.157565e-2 + y2*(0.916281e-2 + y2*(
+            -0.2057706e-1 + y2*(0.2635537e-1 + y2*(-0.1647633e-1 + y2*0.392377e-2))))))))
+    return conditional(le(ax, 3.75), expr1, expr2)
+
+
+def bessk0(x):
+    """
+    Modified Bessel function of the second kind. Code taken from [3].
+    """
+    y1 = x*x/4.0
+    expr1 = -ln(x/2.0)*bessi0(x) + (-0.57721566 + y1*(0.42278420 + y1*(
+        0.23069756 + y1*(0.3488590e-1 + y1*(0.262698e-2 + y1*(0.10750e-3 + y1*0.74e-5))))))
+    y2 = 2.0/x
+    expr2 = exp(-x)/sqrt(x)*(1.25331414 + y2*(-0.7832358e-1 + y2*(0.2189568e-1 + y2*(
+        -0.1062446e-1 + y2*(0.587872e-2 + y2*(-0.251540e-2 + y2*0.53208e-3))))))
+    return conditional(ge(x, 2), expr2, expr1)
+
+
+class PointDischargeParameters(object):
+    """
+    Problem parameter class, including point source representation.
+
+    Delta functions are not in H1 and hence do not live in the FunctionSpaces we
+    seek to use. Here we use a Gaussian approximation with a small radius. The
+    small radius has been calibrated against the analytical solution. See [2]
+    for details.
+    """
+    def __init__(self, offset):
+        self.offset = offset
+
+        # Physical parameters
+        self.diffusivity = Constant(0.1)
+        self.viscosity = None
+        self.drag = Constant(0.0025)
+        self.uv = Constant(as_vector([1.0, 0.0]))
+        self.elev = Constant(0.0)
+
+        # Stabilisation
+        self.use_lax_friedrichs_tracer = True
+
+        # Parametrisation of point source
+        self.source_x, self.source_y = 2.0, 5.0
+        self.source_r = 0.05606298 if self.use_lax_friedrichs_tracer else 0.05606298
+        self.source_value = 100.0
+
+        # Boundary conditions
+        self.boundary_conditions = {
+            'tracer': {
+                1: {'value': Constant(0.0)},      # inflow
+                2: {'open': None},                # outflow
+                3: {'diff_flux': Constant(0.0)},  # Neumann
+                4: {'diff_flux': Constant(0.0)},  # Neumann
```

Comment:
I think this is equivalent to omitting the boundary condition declaration (which implies that the boundary flux term is omitted).